### PR TITLE
Bugs and Improvements

### PR DESCRIPTION
**Pull request by zpawn**
Story https://app.shortcut.com/deskpro/story/98534/hubspot-bugs-and-improvements

* Set default to current datetime when logging activity
* Unlink the contact if it's not in HubSpot
* Save metadata in deskpro when linking/creating/updating a contact
* Fixed TS error with ErrorBoundary component

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -89,32 +63,29 @@ function App() {
     return (
         <Suspense fallback={<LoadingSpinner/>}>
             <QueryErrorResetBoundary>
-                {({ reset }) => {
-                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-                    // @ts-ignore
```
Comment (amDavid01) [inline src/App.tsx:94]: could you just leave a comment where you ignore ts issues mentioning why? Better for other devs to see aswell

File: src/pages/LinkPage.tsx [hunk 2]
```diff
@@ -94,7 +97,10 @@ const LinkPage: FC = () => {
         }
 
         setLoading(true);
-        setEntityContact(client, deskproUser.id, selectedContactId)
+        getContactInfo(selectedContactId)
+            .then((data) => {
+                return setEntityContact(client, deskproUser.id, selectedContactId, getEntityMetadata(data));
+            })
             // eslint-disable-next-line @typescript-eslint/ban-ts-comment
             // @ts-ignore
```
Comment (amDavid01) [inline src/pages/LinkPage.tsx:105]: same here

File: src/App.tsx (hunk 1 above)
Comment (zpawn) [reply src/App.tsx:94]: Update `@types/react` and `@types/react-dom` to fix this ts error

File: src/pages/LinkPage.tsx (hunk 2 above)
Comment (zpawn) [reply src/pages/LinkPage.tsx:105]: need to fix types in the `app-sdk`

Comment (zpawn) [reply src/pages/LinkPage.tsx:105]: OK, I'll leave it

File: src/App.tsx (hunk 1 above)
Comment (amDavid01) [reply src/App.tsx:94]: no I mean in the code itself
//Update @types/react and @types/react-dom to fix this ts error
//eslint-disable etc..
//@ts-ignore

Comment (zpawn) [reply src/App.tsx:94]: Yes, I understood you. I will leave it for the future

**Review by amDavid01 — APPROVED** [review]

**Review by AshleyDawson — APPROVED** [review]

**Pull request merged**